stdlib/InteractiveUtils/src/editless.jl

```diff
@@ -220,7 +220,7 @@ method to edit. For modules, open the main source file. The module needs to be l
 To ensure that the file can be opened at the given line, you may need to call
 `define_editor` first.
 """
-edit(f)                   = edit(functionloc(f)...)
+edit(f)                   = length(methods(f)) > 1 ? methods(f) : edit(functionloc(f)...)
```

Comment:
It might be nice to print out a bit more information alongside this — both the fact that `f` has multiple methods and detailing the control-q trick.

Reply:
Great idea! Will add. Btw, two questions:
1. Should I specify `edit(f::Function)`, since we are passing to `methods()`, and leave the current generic `edit(f)` signature - since `functionloc()` has other signatures?
2. Also, can I add another method allowing an index, without the #[^Q]?
```julia
edit(f::Function, idx::Integer) = edit(methods(f).ms[idx])
```
Sometimes I edit a function, then close, then want to edit again. Having `edit(f, idx)` would allow one to just re-run the command, without needing to get the methods and then `#<c-q>` again. Thoughts?

I also find the `#<c-q>` a bit of awkward finger movements (hard to explain why, but if you try pushing "5" and then "ctl-q" with the left control button, you might also find the pinky finger pushing harder than usual) compared to `<up><backspace>,#)<enter>`, though this is more a personal preference.

Reply:
Another alternative — since we're already interactive  — would be to just display one of the REPL's builtin interactive menus.  Of course that's more involved, but that'd be ideal.

Reply:
Like this?
```julia
function _edit(f)
    if length(methods(f).ms) > 1
        menu = RadioMenu([string(m) for m in methods(f).ms], pagesize=4)
        choice = request("Choose a signature to edit:", menu)
        methods(f).ms[choice]
    else
        edit(functionloc(f)...)
    end
end

```
However, I'd personally be in favor of just using the `methods()` return because the interactive menus seem to break tmux/screen terminal managers (the output doesn't clear, just keeps printing new lines during scrolling). But returning methods seems a bit more robust.

Reply:
```suggestion
function edit(@nospecialize f)
    ms = methods(f)
    length(m) == 1 && edit(functionloc(m[1])...)
    length(m) > 1 && return ms
    lenth(m) == 0 && functionloc(f) # throws
    nothing
end
edit(@nospecialize(f), idx::Integer) = edit(methods(f).ms[idx])
```

I agree with adding #2 also, so I've added that to the suggestion too

Reply:
Thanks, sounds good! I have added your suggestions to the PR.